mergify_engine/queue/queue_freeze.py

```diff
@@ -0,0 +1,130 @@
+        async for key, qf_raw in repository.installation.redis_queue.hscan_iter(
+            name=cls._get_redis_hash(repository),
+            match=cls._get_redis_key_match(repository),
+        ):
+            qf = msgpack.unpackb(qf_raw, timestamp=3)
+            queue_freezes.append(
+                queues.QueueFreeze(
+                    name=qf["name"],
+                    reason=qf["reason"],
+                    user_app_name=qf["user_app_name"],
+                    freeze_date=qf["freeze_date"],
+                )
+            )
+
+        return queue_freezes
+
+    @classmethod
+    async def get(
+        cls, repository: context.Repository, queue_name: str
+    ) -> typing.Optional["QueueFreeze"]:
+
+        queue_freeze_data_raw = await repository.installation.redis_queue.hget(
+            cls._get_redis_hash(repository),
+            cls._get_redis_key(repository, queue_name),
+        )
+
+        if queue_freeze_data_raw is not None:
+            # NOTE(Syffe): timestamp parameter means that timestamp variables will be converted to
+            # datetime (value 3=to_datetime()). Other values can be used: 1=to_float(), 2=to_unix_ns()
+            queue_freeze_data = msgpack.unpackb(queue_freeze_data_raw, timestamp=3)
+
+            return cls(
+                repository=repository,
+                name=queue_freeze_data["name"],
+                reason=queue_freeze_data["reason"],
+                user_app_name=queue_freeze_data["user_app_name"],
+                freeze_date=queue_freeze_data["freeze_date"],
+            )
+
+        else:
+            return None
+
+    @classmethod
+    def _get_redis_hash(cls, repository: context.Repository) -> str:
+        return f"merge-freeze~{repository.installation.owner_id}"
+
+    @classmethod
+    def _get_redis_key(cls, repository: context.Repository, queue_name: str) -> str:
+        return f"{repository.repo['id']}~{queue_name}"
+
+    @classmethod
+    def _get_redis_key_match(cls, repository: context.Repository) -> str:
+        return f"{repository.repo['id']}~*"
+
+    async def save(self) -> None:
+
+        await self.repository.installation.redis_queue.hset(
+            self._get_redis_hash(self.repository),
+            self._get_redis_key(self.repository, self.name),
+            msgpack.packb(
+                {
+                    "name": self.name,
+                    "reason": self.reason,
+                    "user_app_name": self.user_app_name,
+                    "freeze_date": self.freeze_date,
+                },
+                # NOTE(Syffe): datetime parameter means that datetime variables will be converted to a timestamp
+                # in order to be serialized
+                datetime=True,
+            ),
+        )
+
+    async def delete(self) -> None:
+        await self.repository.installation.redis_queue.hdel(
+            self._get_redis_hash(self.repository),
+            self._get_redis_key(self.repository, self.name),
+        )
```

Comment:
```suggestion
    async def delete(self) -> bool:
        return bool(await self.repository.installation.redis_queue.hdel(
            self._get_redis_hash(self.repository),
            self._get_redis_key(self.repository, self.name),
        ))
```
`hdel` returns `1` if the deletion happened